core/tests/protractor/accessibility.js

```diff
@@ -41,6 +120,164 @@ describe('screenreader and keyboard user accessibility features', function() {
+      await libraryPage.get();
+      await browser.actions().sendKeys('c').perform();
+      var categoryBar = element(by.css(
+        '.protractor-test-search-bar-dropdown-toggle'));
+      expect(await categoryBar.getAttribute('id')).toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus away from the category bar.
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      expect(await categoryBar.getAttribute('id')).not.toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus to the category bar.
+      await browser.actions().sendKeys('c').perform();
+      expect(await categoryBar.getAttribute('id')).toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+    });
+
+  it('should move focus to skip to main content button in exploration player',
+    async function() {
+      // Should Create a user and login.
+      await users.createUser('user11@accessibility.com', 'user11accessibility');
+      await users.login('user11@accessibility.com', true);
+
+      // Should create and play a dummy exploration.
+      await createDummyExplorationOnDesktop();
+      await libraryPage.get();
+      await libraryPage.findExploration('Dummy Exploration');
+      await libraryPage.playExploration('Dummy Exploration');
+
+      // Should move the focus to the skip to main content button.
+      await browser.actions().sendKeys('s').perform();
+      var skipButton = element(by.css('.protractor-test-skip-link'));
+      expect(await skipButton.getAttribute('id')).toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus away from the skip to main content button.
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      expect(await skipButton.getAttribute('id')).not.toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus back to the skip to main content button.
+      await browser.actions().sendKeys('s').perform();
+      expect(await skipButton.getAttribute('id')).toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+    });
+
+  it('should move focus to next and back buttons in exploration player',
+    async function() {
+      // Play a dummy exploration.
+      await libraryPage.get();
+      await libraryPage.findExploration('Dummy Exploration');
+      await libraryPage.playExploration('Dummy Exploration');
+      await explorationPlayerPage.submitAnswer('Continue', null);
+      await waitFor.pageToFullyLoad();
```

Comment:
Make sure a loading indicator is actually shown here, otherwise this won't do what you expect